float64[] points
---
time stamp
float64[] gradients
float64[] distances
bool[] in_bounds